mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/parser/Combinators.kt

```diff
@@ -118,4 +118,14 @@ object Combinators {
    */
   fun <I, O> Parser<I, O>.repeatAtLeast(count: Int): Parser<I, List<O>> =
       repeat().filter { it.size >= count }
+
+  /**
+   * Return a [Parser] which doesn't contains any null value
+   *
+   * @param I the type of the input.
+   * @param O the type of the output.
+   * @return a [Parser] without null value
+   */
+  @Suppress("UNCHECKED_CAST")
+  fun <I, O> Parser<I, O?>.filterNotNull(): Parser<I, O> = filter { it != null }.map { it as O }
```

Comment:
Probably better to not have the @Suppress("UNCHECKED_CAST")
```suggestion
fun <I, O> Parser<I, O?>.filterNotNull(): Parser<I, O> =
      filter { it != null }.map { checkNotNull(it) }
```